### PR TITLE
Remove secrets in PSAzurerRmAccount from display

### PR DESCRIPTION
**Pull request by msJinLei**
The PR is to fix https://github.com/Azure/azure-powershell/issues/15427

Service principal secret and service principal certificate are required for service principal user to access Azure resources.
Previously PowerShell can access the plain text values of service principal secret and service principal certificate password using 
```
- (Get-AzContext).Accounts.ExtendedProperties.ServicePrincipalSecret
- (Get-AzContext).Accounts.ExtendedProperties.CertificatePassword
```

To improve the security, the PR removes the these 2 attributes from `(Get-AzContext).Accounts`, whose type is `PSAzurerRmAccount`.

After the change, the context object retrieved by Get-AzContext no longer has ServicePrincipalSecret and CertificatePassword. The cmdlet Set-AzContext, Rename-AzContext accept context object as the input, which brings the risk that the context without  ServicePrincipalSecret and CertificatePassword will be added into context list and even be set as current context and then break the current environment.

To solve the issue, the PR also changes the logic when new context object is added to the context list. Firstly find whether there is an existing object with the same account id, tenant and subscription. If found, merge the attributes in the existing one but absent in the new one to the new one and then add the new one to the context list.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Accounts/Accounts/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 -->
 
 ## Upcoming Release
+* Removed `ServicePrincipalSecret` and `CertificatePassword` in Account of Context from display [#15427]
```
Comment (dingmeng-xue) [inline src/Accounts/Accounts/ChangeLog.md:22]: from display or we have removed them from Azure Context psobject? I remember it is latter.

File: src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs [hunk 2]
```diff
@@ -488,11 +488,25 @@ public bool TryRenameContext(string sourceName, string targetName)
...
         public bool TrySetContext(string name, IAzureContext context)
         {
             bool result = false;
             if (Contexts != null)
             {
+                string oldName = null;
+                if (TryFindContext(context, out oldName))
+                {
+                    var oldContext = Contexts[oldName];
+                    oldContext.Update(context);
```
Comment (dingmeng-xue) [inline src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs:507]: `context` is from user's import. Is it possible this line will change `oldContext` in `contexts`?

File: src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs [hunk 3]
```diff
@@ -488,11 +488,25 @@ public bool TryRenameContext(string sourceName, string targetName)
...
+        /// </summary>
+        /// <param name="name">The specified new name of the context.</param>
+        /// <param name="context">The new context to set as default.</param>
         public bool TrySetContext(string name, IAzureContext context)
         {
             bool result = false;
             if (Contexts != null)
             {
+                string oldName = null;
+                if (TryFindContext(context, out oldName))
```
Comment (dingmeng-xue) [inline src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs:504]:
```suggestion
                if (TryFindContext(context, out string oldName))
```

File: src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs (hunk 2 above)
Comment (msJinLei) [reply src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs:506]: Should use deepCopy. Fixed

File: src/Accounts/Accounts/ChangeLog.md (hunk 1 above)
Comment (msJinLei) [reply src/Accounts/Accounts/ChangeLog.md:22]: > 

There are 2 classes that relate to Account of Context in Az.Accounts, AzureAccount is for internal use and PSAzureRmAccount is for powershell operation. The PR just removes `ServicePrincipalSecret` and `CertificatePassword` from PSAzureRmAccount.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**